wp_serde_helper/src/wp_serde_date.rs

```diff
@@ -17,15 +18,35 @@ pub mod wp_utc_date_format {
     where
         D: Deserializer<'de>,
     {
-        let s = String::deserialize(deserializer)?;
+        let value = serde_json::Value::deserialize(deserializer)?;
```

Comment:
I don't think `serde_json` should be used here? The original function call seems to be more correct?

Reply:
I assume you mean it can be deserialized as `u64` first and if it errors then we try deserializing it as `String`? That's probably a little better, but unless there is a big performance issue, I think it's fine to parse it into `serde_json::Value` first and then see if it can be cast to other types as it's implemented in this PR. 🤷‍♂️

Reply:
Sorry, I was not clear. `deserializer` is not always serde_json's Deserializer. Hypothetically speaking, if `WpGmtDateTime` is serialised into other forms (binary, XML, whatever), this code may break, because the data it tries to deserialize may not be in JSON string format.

Reply:
Ah.. right. 🤦

Reply:
Addressed in d776c9c2e34334e8a383f94bcc8b718a0d788f63 – thanks for the suggestion @crazytonyli!